kraken-build/tests/kraken_std/integration/cargo/test_cargo_private_registry.py

```diff
@@ -77,6 +77,26 @@ def publish_lib_and_build_app(repository: CargoRepositoryWithAuth, tempdir: Path
             )
             project1.context.execute(["fmt", "lint", "publish"])
 
+        # Try to publish the same version again - the task should be skipped and should not fail
+        with kraken_ctx() as ctx, kraken_project(ctx) as project2:
+            project2.directory = lib_dir
+            cargo_registry(
+                cargo_registry_id,
+                repository.index_url,
+                read_credentials=repository.creds,
+                publish_token=repository.token,
+            )
+            cargo_auth_proxy()
+            task = cargo_sync_config()
+            task.git_fetch_with_cli.set(True)
+            cargo_check_toolchain_version(minimal_version="1.60")
+            cargo_publish(
+                cargo_registry_id,
+                version=publish_version,
+                cargo_toml_file=project1.directory.joinpath("Cargo.toml"),
+            )
+            project1.context.execute(["fmt", "lint", "publish"])
```

Comment:
We should assert that the task status indicates that it was skipped. One way to do this would be to make `Context.execute()` return the `TaskGraph`, and then use `TaskGraph.get_status(cargo_publish_task)`.

Reply:
Will do - also noticed that this should be using `project2`. Oops